source/driver-vs-mongoose.txt

```diff
@@ -0,0 +1,142 @@
+level. In comparison, Mongoose applies schema validation at the application level. In the
+{+driver-short+} schema validation works on individual collections by
+using either `JSON Schema Validation <https://www.mongodb.com/docs/manual/core/schema-validation/specify-json-schema/>`__
+or `Schema Validation with Query Operators <https://www.mongodb.com/docs/manual/core/schema-validation/specify-query-expression-rules/>`__.
+Validation with query operators is useful for dynamic validation rules
+calculated at runtime. By contrast, JSON validation is useful for, among other
+reasons, being an open standard which makes it easy to connect to
+third-party libraries.
+
+Similarities
+------------
+
+As Mongoose is built upon the {+driver-long+}, they share many
+capabilities. Even if you use Mongoose, you always have the option to
+use any native function or command built into the {+driver-short+}
+alongside Mongoose. Mongoose also offers its own methods that are quite
+similar to the {+driver-short+}, such as ``updateOne()`` and
+``replaceOne()``.
+
+.. tip::
+   
+   There are many other Mongoose specific methods and APIs not discussed. For
+   more information on how to use Mongoose, check out the official `Mongoose
+   documentation <https://mongoosejs.com/docs/>`__.
+
+Additional Functionality Provided by Mongoose
+---------------------------------------------
+
+On top of schema management, there is a multitude of additional
+functionality provided by Mongoose. In this section, we will go over a
+few of the additional functionalities.
+
+Data Modeling
+~~~~~~~~~~~~~
+
+Mongoose introduces the concept of models, which are JavaScript classes
+defined based on schemas. Models act as an interface to interact
+with MongoDB collections, abstracting away the complexities of database
+operations.
+
+Middleware Hooks
+~~~~~~~~~~~~~~~~
+
+Mongoose supports middleware hooks, both pre and post-hooks, that allow you
+to add custom logic before or after certain database operations like
+save, update, and remove. This is useful for implementing additional
+actions or validations. Mongoose supports adding custom plugins and
+middleware functions to extend its functionality further. 
+
+Virtuals
+~~~~~~~~
+
+In Mongoose, a virtual is a property that is not stored in MongoDB.
+Virtuals are typically used for computed properties on
+documents. Virtuals are additional fields for a given model; their values
+can be set manually or automatically with defined functionality. They
+only exist logically and are not written to the document, collection, or
+database.
+
+
+Conclusion
+----------
+
+Both Mongoose and the {+driver-long+} offer comparable features
+and capabilities. Mongoose helps ease MongoDB development for
+newcomers while the {+driver-long+} seamlessly integrates with your MongoDB
+database, providing an exceptional and adaptable development experience.
+By leveraging the driver, you enable your database to excel at its core
+functionality.
```

Comment:
```suggestion
and capabilities. Mongoose can ease the development of applications for
new users of MongoDB. Because the {+driver-short+} is actively maintained and improved by MongoDB, 
the driver seamlessly integrates with your MongoDB
database, providing a straightforward and adaptable development experience.
```